x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/Mapper.java

```diff
@@ -104,6 +106,46 @@ public PhysicalPlan map(LogicalPlan p) {
         //
         // Unary Plan
         //
+        if (localMode == false && p instanceof Enrich enrich && enrich.mode() == Enrich.Mode.REMOTE) {
+            // When we have remote enrich, we want to put it under FragmentExec, so it would be executed remotely.
```

Comment:
remote typically implies remote cluster. I think you mean data node (or in ESQL terminology local as in local planning).

Reply:
Yes, here it is meant that remote enrich would go to the remote cluster. It may also do it effectively on the local cluster if the enrich policy & indexes are there, but the important part I think is that it'd also go to the remote.